### PR TITLE
DUOS-1438[risk=no] Add Demoable Data Custodian Subsection to SO Console

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [DUOS-1438](https://broadworkbench.atlassian.net/browse/DUOS-1438)

PR adds a Data Custodian sub-component to the SO Console that allows a user to issue/revoke for demo purposes. This means that while the UI will have state updates, there are no updates being saved to the back-end (which means perceived updates will not persist). Demo is only for the SO Console, which means Admin functionality is not available.

This is not meant to be the final version of the sub-component, however the demo will provide a good jumping off point once Data Custodian work begins.
 
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/SigningOfficialConsole.js [hunk 1]
```diff
@@ -73,13 +81,35 @@ export default function SigningOfficialConsole(props) {
...
+            color: 'blue',
+            fontSize: '1.8rem',
+            componentType: tabs.researcher,
+            setSelected: setSelectedTag,
+            selectedType: selectedTag
+          })
+        ]),
+        h(SigningOfficialTable, {isRendered: selectedTag === tabs.researcher, researchers, signingOfficial, unregisteredResearchers, isLoading}, []),
+        h(DataCustodianTable, {isRendered: selectedTag === tabs.custodian, researchers, signingOfficial, unregisteredResearchers, isLoading}, [])
+      ]),
```
Comment (rushtong) [inline src/pages/SigningOfficialConsole.js]: Can we wrap the new content in a conditional on env? Look at https://github.com/DataBiosphere/duos-ui/blob/develop/src/libs/config.js#L6 which should give you the correct environment. See also:
* https://duos-k8s.dsde-dev.broadinstitute.org/config.json
* https://duos-k8s.dsde-staging.broadinstitute.org/config.json
* https://duos-k8s.dsde-prod.broadinstitute.org/config.json

File: src/components/modals/DataCustodianFormModal.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      getOptionLabel: (option) =>
+        `${option.displayName || 'New User'} (${
+          option.email || 'No email provided'
+        })` || option.email,
+    }),
+  ]);
+  return div({ display: 'flex' }, [template]);
+};
+
+export default function DataCustodianFormModal(props) {
```
Comment (rushtong) [inline src/components/modals/DataCustodianFormModal.js:74]: Not really feedback for this PR, but just wanted to express my hopes that we can address the modal styling at some point. Will dig around and see if we have any stories in the backlog for this task.

**Review by rushtong — COMMENTED** [review]
Overall, this looks good. One minor request inline (env-check) and then I'd be happy with merging.

File: src/pages/SigningOfficialConsole.js (hunk 1 above)
Comment (JVThomas) [reply src/pages/SigningOfficialConsole.js]: What's the conditional? Don't render on production?

Comment (rushtong) [reply src/pages/SigningOfficialConsole.js]: Yes, exactly. Staging and dev should be fine.

**Review by rushtong — APPROVED** [review]
Thank you for the changes! Pending tests, 👍🏽

**Pull request merged**